src/lib/scheduler/operator_task.cpp

```diff
@@ -45,6 +45,44 @@ std::shared_ptr<OperatorTask> add_operator_tasks_recursively(const std::shared_p
   return task;
 }
 
+/**
+ * Sets tasks that can be used to prune chunks by predicates with uncorrelated subqueries as successors of the GetTable
+ * tasks. Guarantees that the resulting task graph is still acyclic.
+ */
+void link_tasks_for_subquery_pruning(const std::unordered_set<std::shared_ptr<OperatorTask>>& tasks) {
+  for (const auto& task : tasks) {
+    const auto& op = task->get_operator();
+    if (op->type() != OperatorType::GetTable) {
+      continue;
+    }
+
+    const auto& get_table = static_cast<GetTable&>(*op);
+    for (const auto& table_scan : get_table.prunable_subquery_scans()) {
+      for (const auto& subquery : table_scan->uncorrelated_subqueries()) {
+        // All tasks have already been created, so we must be able to get the cached task from each operator.
```

Comment:
they are now created since we created the task graph now. That's why we do this now and not during task creation. Basically, the same as for operator/node copy and LQP translator